### DISCUSSION TITLE
Which components can pose accessibility challenges?

### DISCUSSION THREAD

**Discussion by amycole501** [uswds/uswds#5759] [Accessibility]
USWDS components have different levels of complexity. Some have a single interaction — just a click or scroll, for example. Others have multiple interactions. Some have no interaction at all. Other design decisions, like colors and fonts, can also complicate a component’s accessibility. 

We want to know about your experience with USWDS component accessibility. Which components can pose accessibility challenges? Let us know:

- Do certain components pose greater accessibility challenges than others? How and for whom?
- Have you found a solution to a component accessibility issue that you’d like to share?

**Comment by shkeating** [comment]
**combobox** is troublesome and I am at a point where I just advocate against using it altogether because I don't know what a good solution is

**Reply by amycole501** [reply]
Agreed, the combobox is often a trouble spot especially for assistive technology users in our user testing. I've noticed comboboxes are an issue on many other websites not just ones that use USWDS so it's pervasive. Hopefully our testing will reveal opportunities for improvement we can implement and share with teams!

**Reply by amycole501** [reply]
@shkeating Do you have any examples of a combobox you or your team has encountered that works well? Or are they all consistently poor performers?

**Comment by shkeating** [comment]
**duplicate accessible names**. If people take a **Search** or a **Date Picker** out of the box and use multiple of the same component on the same page, they will end up with issues around having non unique accessible names for these buttons that can be troublesome for screen reader users.

**Reply by amycole501** [reply]
I've seen this happen on forms which ask a person to put a current and past address. Unique labels are critical to ensure a person knows which field they're interacting with (or which field has an error if error handling is used).

**Comment by shkeating** [comment]
**guidance around language usage** - for components showing alternate options in differnent languagees (for example **Search** having a spanish variant), there should be info in the accessibility section reminding folks to set their page language in their html `<head>` or if they are only using a language different from the rest of the page in one area there is a `lang` attribute assigned programmatcally to that area

**Reply by mejiaj** [reply]
Good idea, I've created an issue for this on the website repo - uswds/uswds-site#2517.
